lib/beaker-pe/install/pe_utils.rb

```diff
@@ -390,16 +390,23 @@ def fetch_pe_on_unix(host, opts)
             if host['platform'] =~ /eos/
               host.get_remote_file("#{path}/#{filename}#{extension}")
             else
-              unpack = 'tar -xvf -'
-              unpack = extension =~ /gz/ ? 'gunzip | ' + unpack  : unpack
               if opts[:fetch_local_then_push_to_host]
                 fetch_and_push_pe(host, path, filename, extension)
                 command_file_push = 'cat '
               else
                 curlopts = opts[:use_proxy] ? "--proxy #{opts[:proxy_hostname]}:3128 " : ""
-                command_file_push = "curl -L #{curlopts}#{path}/"
+                command_file_push = "curl --fail --location --output #{filename}#{extension} #{curlopts}#{path}/"
               end
-              on host, "cd #{host['working_dir']}; #{command_file_push}#{filename}#{extension} | #{unpack}"
+
+              retry_requirements = {
+                desired_exit_codes: [0],
+                max_retries: 3,
+                verbose: 'true'
+              }
+              fetch_tarball_command = "cd #{host['working_dir']}; #{command_file_push}#{filename}#{extension}"
+              retry_on(host, fetch_tarball_command, retry_requirements)
+              on host, "cd #{host['working_dir']}; tar -xvf #{filename}#{extension}"
```

Comment:
The only thing that comes to mind is the question of whether or not we can rely on tar being able to gunzip automagically on all platforms this method would execute on. But since I think this would only be used on pe master platforms, I don't think that's an issue.

Reply:
I went through exactly that same series of steps. I'm willing to fix it if I got it wrong.